types/parse/index.d.ts

```diff
@@ -79,6 +80,11 @@ declare enum ErrorCode {
 
 declare namespace Parse {
 
+    // Helper types for extracting an extended generic type from an
+    // (Parse.)Object or Array.
+    type Unarray<T> = T extends Array<infer U> ? U : T;
+    type Unobject<X> = X extends Object<infer U> ? U extends any ? any : U : X;
```

Comment:
`Unobject` of something that isn't an `Object` should `never` happen.

```suggestion
    type Unobject<X> = X extends Object<infer U> ? U : never;
```